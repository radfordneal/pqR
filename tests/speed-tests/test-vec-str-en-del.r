source("time.r")
 
test.name <- "vec-str-en-del.enlarge"

f <- test.cmp (function (n,v)
{ m <- length (v) 
  v1 <- v[[1]]
  for (i in 1:n) v[[m+i]] <- v1
  v
})

v <- rep(list(1.2,4.9),10000)
sys.time(r <- f(10000,v))
print(c(length(r),sum(unlist(r))))

v <- rep(c("1.2","4.9"),10000)
sys.time(r <- f(10000,v))
print(c(length(r),sum(as.numeric(r))))


test.name <- "vec-str-en-del.delete"

f <- test.cmp (function (n,v)
{ for (i in 1:n) v[[100+i]] <- NULL
  v
})

v <- rep(list(1.2,4.9),10000)
sys.time(r <- f(8000,v))
print(c(length(r),sum(unlist(r))))

f <- test.cmp (function (n,v)
{ for (i in 1:n) v[100+i] <- NULL
  v
})

v <- rep(list(1.2,4.9),10000)
sys.time(r <- f(8000,v))
print(c(length(r),sum(unlist(r))))

v <- rep(list(1.2,4.9),10000)
names(v) <- rep("abc",10000)
sys.time(r <- f(8000,v))
print(c(length(r),sum(unlist(r))))

f <- test.cmp (function (n,v)
{ for (i in 1:n) v[1000:1002] <- NULL
  v
})

v <- rep(list(1.2,4.9),10000)
sys.time(r <- f(6000,v))
print(c(length(r),sum(unlist(r))))


test.name <- "vec-str-en-del.str-cat"

f <- test.cmp (function (n,s1,s2,s3)
{
  for (i in 1:n) v <- c(s1,s2,s3)
  v
})

s1 <- rep("1.2",1000)
s2 <- rep("3.4",1001)
s3 <- rep("5.6",1001)

sys.time(r <- f(50000,s1,s2,s3))
print(c(length(r),sum(as.numeric(r))))

# correctness checks.

a <- c("abc","def","ghi","xxx")
b <- a
a[[6]] <- "yyy"
print(c(a,b))

a <- list(a=3,b=T,c="aa",d=list(1),e=4L,f=3,g=90)
b <- a
b[["b"]] <- NULL
print(b)

b[2:3] <- NULL
print(b)

b[c(1,4)] <- NULL
print(b)
