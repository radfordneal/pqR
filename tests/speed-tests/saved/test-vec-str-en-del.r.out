
---- vec-str-en-del.enlarge 

Time:
[1] 30000 73000
Time:
[1] 30000 73000

---- vec-str-en-del.delete 

Time:
[1] 12000 51400
Time:
[1] 12000 51400
Time:
[1] 12000 51400
Time:
[1] 2000 6100

---- vec-str-en-del.str-cat 

Time:
[1]  3002 10209
 [1] "abc" "def" "ghi" "xxx" NA    "yyy" "abc" "def" "ghi" "xxx"
$a
[1] 3

$c
[1] "aa"

$d
$d[[1]]
[1] 1


$e
[1] 4

$f
[1] 3

$g
[1] 90

$a
[1] 3

$e
[1] 4

$f
[1] 3

$g
[1] 90

$e
[1] 4

$f
[1] 3

